### ISSUE TITLE
`@pulumiverse/vercel` provider fails to create DNS record after upgrading from v3.3.29 to v3.4.28

### ISSUE THREAD

**Issue by waltershewmake** [anomalyco/sst#5253]
Hi, after I updated SST from v3.3.29 to the latest version, v3.4.28, the vercel DNS provider stopped working for deployments. Downgrading back to v3.3.29 solves the issue. Here is the error message I see:

```bash
|  Error       Api sst:aws:Service → ApiSslCAARecordApidevnessylcom vercel:index:DnsRecord
|  Error creating DNS Record: Could not create DNS Record, unexpected error: invalid_value - Invalid record value: "0 issue amazonaws.com".
|  Created     Api sst:aws:Service → ApiLoadBalancerSecurityGroup aws:ec2:SecurityGroup (2.5s)

✕  Failed    
   Api sst:aws:Service → ApiSslCAARecordApidevnessylcom vercel:index:DnsRecord
   Error creating DNS Record: Could not create DNS Record, unexpected error: invalid_value - Invalid record value: "0 issue amazonaws.com".
   ```

Here is the relevant infrastructure definition:

```typescript
import { rds } from './database';
import {
  ClerkPublishableKey,
  ClerkSecretKey,
  ClerkWhsec,
  GoogleMapsApiKey,
} from './secrets';
import { vpc } from './vpc';

export const apiCluster = new sst.aws.Cluster('ApiCluster', {
  vpc,
});

export const api = apiCluster.addService('Api', {
  dev: {
    url: 'http://localhost:3333',
  },
  architecture: 'arm64',
  containers: [
    {
      name: 'hono',
      dev: {
        command: 'pnpm dev',
        directory: './apps/api',
      },
      image: {
        dockerfile: './apps/api/Dockerfile',
      },
    },
  ],
  loadBalancer: {
    domain: {
      name: `api.${$app.stage}.nessyl.com`,
      dns: sst.vercel.dns({
        domain: 'nessyl.com',
      }),
    },
    ports: [
      { listen: '80/http', redirect: '443/https' },
      { listen: '443/https', forward: '3333/http' },
    ],
  },
  cpu: '2 vCPU',
  memory: '4 GB',
  link: [
    rds,
    GoogleMapsApiKey,
    ClerkSecretKey,
    ClerkPublishableKey,
    ClerkWhsec,
  ],
});
```

**Comment by waltershewmake** [comment]
Ok, I narrowed it down to `v3.4.8`. The vercel provider works correctly in version `v3.4.7` but not `v3.4.8`. See 0136acb40df45282e1dcb0346624cc6e2d736a4a for details

**Comment by mharrvic** [comment]
@waltershewmake Thanks for digging! also, the `v3.4.7` works on my end (even the latest version [3.4.30](https://www.npmjs.com/package/sst/v/3.4.30) doesn't work). Hope they fix this in future release (i can look into this as well if i have some extra time!)

**Comment by jayair** [comment]
Thanks for digging into it! I'll ask the team to take a look. If anybody creates a PR in the meantime, that works too.

**Comment by fwang** [comment]
Fixed in v3.4.36

@waltershewmake Thanks for tracking it down. The issue was the missing quotes around the record value.
